### PR TITLE
Added Pod DNS Chaos

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

- This PR adds Pod DNS chaos

### CODE REVIEW

File: pkg/generic/pod-dns-chaos/environment/environment.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	"strconv"
+
+	experimentTypes "github.com/litmuschaos/litmus-go/pkg/generic/pod-dns-chaos/types"
+	"github.com/litmuschaos/litmus-go/pkg/types"
+	clientTypes "k8s.io/apimachinery/pkg/types"
+)
+
+//GetENV fetches all the env variables from the runner pod
+func GetENV(experimentDetails *experimentTypes.ExperimentDetails) {
+	experimentDetails.ExperimentName = Getenv("EXPERIMENT_NAME", "")
```
Comment (ispeakc0de) [inline pkg/generic/pod-dns-chaos/environment/environment.go:14]:
```suggestion
	experimentDetails.ExperimentName = Getenv("EXPERIMENT_NAME", "pod-dns-chaos")
```

File: experiments/generic/pod-dns-chaos/experiment/pod-dns-chaos.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		"Label":     experimentsDetails.AppLabel,
+		"Ramp Time": experimentsDetails.RampTime,
+	})
+
+	// Calling AbortWatcher go routine, it will continuously watch for the abort signal and generate the required events and result
+	go common.AbortWatcher(experimentsDetails.ExperimentName, clients, &resultDetails, &chaosDetails, &eventsDetails)
+
+	//PRE-CHAOS APPLICATION STATUS CHECK
+	log.Info("[Status]: Verify that the AUT (Application Under Test) is running (pre-chaos)")
+	err = status.CheckApplicationStatus(experimentsDetails.AppNS, experimentsDetails.AppLabel, experimentsDetails.Timeout, experimentsDetails.Delay, clients)
```
Comment (ispeakc0de) [inline experiments/generic/pod-dns-chaos/experiment/pod-dns-chaos.go:74]: Covert this function to status.AUTStatusCheck() -- [refer](https://github.com/litmuschaos/litmus-go/blob/8fad404e570bff723060bddebb3874108e398ebe/experiments/generic/container-kill/experiment/container-kill.go#L75)

File: experiments/generic/pod-dns-chaos/experiment/pod-dns-chaos.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+	} else {
+		log.Error("[Invalid]: Please Provide the correct LIB")
+		failStep := "no match found for specified lib"
+		result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+		return
+	}
+
+	//POST-CHAOS APPLICATION STATUS CHECK
+	log.Info("[Status]: Verify that the AUT (Application Under Test) is running (post-chaos)")
+	err = status.CheckApplicationStatus(experimentsDetails.AppNS, experimentsDetails.AppLabel, experimentsDetails.Timeout, experimentsDetails.Delay, clients)
```
Comment (ispeakc0de) [inline experiments/generic/pod-dns-chaos/experiment/pod-dns-chaos.go:126]: same as above

File: chaoslib/litmus/pod-dns-chaos/lib/pod-dns-chaos.go [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+		// Wait till the completion of the helper pod
+		// set an upper limit for the waiting time
+		log.Info("[Wait]: waiting till the completion of the helper pod")
+		podStatus, err := status.WaitForCompletion(experimentsDetails.ChaosNamespace, appLabel, clients, experimentsDetails.ChaosDuration+60, experimentsDetails.ExperimentName)
+		if err != nil || podStatus == "Failed" {
+			common.DeleteHelperPodBasedOnJobCleanupPolicy(experimentsDetails.ExperimentName+"-"+runID, appLabel, chaosDetails, clients)
+			return errors.Errorf("helper pod failed due to, err: %v", err)
+		}
+
+		//Deleting all the helper pod for container-kill chaos
```
Comment (ispeakc0de) [inline chaoslib/litmus/pod-dns-chaos/lib/pod-dns-chaos.go:136]:
```suggestion
		//Deleting all the helper pod for pod-dns-chaos
```

File: chaoslib/litmus/pod-dns-chaos/lib/pod-dns-chaos.go [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+	// Wait till the completion of the helper pod
+	// set an upper limit for the waiting time
+	log.Info("[Wait]: waiting till the completion of the helper pod")
+	podStatus, err := status.WaitForCompletion(experimentsDetails.ChaosNamespace, appLabel, clients, experimentsDetails.ChaosDuration+60, experimentsDetails.ExperimentName)
+	if err != nil || podStatus == "Failed" {
+		common.DeleteAllHelperPodBasedOnJobCleanupPolicy(appLabel, chaosDetails, clients)
+		return errors.Errorf("helper pod failed due to, err: %v", err)
+	}
+
+	//Deleting all the helper pod for container-kill chaos
```
Comment (ispeakc0de) [inline chaoslib/litmus/pod-dns-chaos/lib/pod-dns-chaos.go:193]:
```suggestion
	//Deleting all the helper pod for pod-dns-chaos
```

File: chaoslib/litmus/pod-dns-chaos/helper/dnschaos.go [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+	return pid, nil
+}
+
+//GetENV fetches all the env variables from the runner pod
+func GetENV(experimentDetails *experimentTypes.ExperimentDetails) {
+	experimentDetails.ExperimentName = Getenv("EXPERIMENT_NAME", "")
+	experimentDetails.AppNS = Getenv("APP_NS", "")
+	experimentDetails.TargetContainer = Getenv("APP_CONTAINER", "")
+	experimentDetails.TargetPods = Getenv("APP_POD", "")
+	experimentDetails.AppLabel = Getenv("APP_LABEL", "")
```
Comment (ispeakc0de) [inline chaoslib/litmus/pod-dns-chaos/helper/dnschaos.go:316]: Do we need AppLabel here?

**Review by ispeakc0de — APPROVED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**